todo/src/main.rs

```diff
@@ -24,33 +30,60 @@ impl Terminal {
         }
     }
 
-    fn ask_new_todo(&mut self) -> Todo {
+    fn ask_new_todo(&mut self) -> Option<Todo> {
         println!("Write your new todo:");
-        Todo::new(self.input())
+        match self.input() {
+            Ok(user_input) => {
+                if user_input.is_empty() {
+                    return None;
+                }
+                Some(Todo::new(user_input))
+            },
+            Err(_) => None
+        }
+ 
     }
 
-    fn show_todo(&mut self, todo: &Todo) {
-        writeln!(self.stdout, "[ ] - {}", todo.message).unwrap()
+    fn show_todo(&mut self, todo: &Todo) -> Result<(), TerminalError> {
+        match writeln!(self.stdout, "[ ] - {}", todo.message) {
+            Ok(_) => Ok(()),
+            Err(err) => Err(TerminalError::Stdout(err))
+        }
     }
 
     fn user_intention(&mut self) -> bool {
         println!("Do you want to input a new todo? (y/n)");
-        self.input() == "y"
+        let user_input = self.input();
+        if let Ok(input) = user_input {
+            return input == "y"
+        } 
+        false
```

Comment:
Normalmente usamos `if else` nessas situações.

```suggestion
        if let Ok(input) = user_input {
            input == "y"
        } else {
            false
        }
```

Reply:
BTW, dá uma olhada no método `map_or` de `Result`.